### DISCUSSION TITLE
How can put the data into an array in laravel blade javascript?

### DISCUSSION THREAD

**Discussion by comprog-ux** [community/community#21453] [Other Feature Feedback, Questions, & Ideas]
<pre><code class="lang-auto">I have these data in table containing:
1 Fält Sundsvall 62.39 17.31 
2 Precision Landskrona 55.87 12.83
outputted by (in the view blade):
@foreach ($data As $i)
{{ $i-&gt;id }} {{ $i-&gt;name }} {{ $i-&gt;contact_city }} {{ $i-&gt;lat }} {{  $i-&gt;lng  }}
@endforeach

I want those  in an array in the &lt;script&gt; area of the blade view to 
populate a google map with markers for each place 
(Sundsvall and Landskrona)
How can I do that?

If I try (in the &lt;script&gt; area where the map builds:
for (i = 1; i &lt; locations; i++) {
var locations = JSON.parse('{{$data}}');

I'm getting (part): '[{&amp;quot;id&amp;quot;:1,&amp;quot;championships_id&amp;quot;:0,&amp;quot;is_public&amp;quot;:0,&amp;quot;
results_is_public&amp;quot;:0,&amp;quot;patrols_is_public&amp;quot;:0,&amp;quot;organizer_id&amp;quot;:73,&amp;quot;
organizer_type&amp;quot;:&amp;quot;App\\Models\\Club&amp;quot;,&amp;quot;invoices_recipient_id&amp;quot;:73,&amp;quot;
invoices_recipient_type&amp;quot;:&amp;quot;App\\Models\\Club&amp;quot;,&amp;quot;name&amp;quot;:&amp;quot;F\u00e4lt&amp;
quot;,&amp;quot;allow_teams&amp;quot;:0,&amp;quot;teams_registration_fee&amp;quot;:0,&amp;quot;website&amp;quot;:&amp;quot;&amp;quot;,&amp;quot;
contact_name&amp;quot;:&amp;quot;&amp;quot;,&amp;quot;contact_venue&amp;quot;:&amp;quot;Sundsvalls skjutbana&amp;quot;.......................

Browser page source, seems to be raw data from the table.</code></pre>

**Comment by billythekid** [comment]
<p>Hi <a class="mention" href="/comprog-ux">@comprog-ux</a>,</p>
<p>I actually have a side-project at <a href="https://ezmap.co">https://ezmap.co</a> (open sourcing the code on this is on my radar) that does exactly this stuff. Built on Laravel and outputting stuff like this in JS.</p>
<p>Laravel escapes certain output by default for safe rendering but in this case you want the actual code to be output.</p>
<p>Try this:</p>
<pre><code class="lang-auto">var locations = JSON.parse('{!! $data !!}')
</code></pre>
<p>reference: <a href="https://laravel.com/docs/8.x/blade#displaying-unescaped-data" class="inline-onebox">Blade Templates - Laravel - The PHP Framework For Web Artisans</a></p>
<p>Depending on how <code>$data</code> is stored (array, collection, iterable etc.) you may need to tweak this but that’s how you don’t output “raw”(unencoded), use <code>{!! $thing !!}</code> instead of <code>{{ $thing }}</code></p>
<p>Let me know how you get on or if you need anything more.</p>
<p>Cheers,<br>
Billy :cowboy_hat_face:</p>

**Comment by billythekid** [comment]
<p>also this bit looks like the lines are the wrong way round, set the variable then access it in the loop</p>
<aside class="quote no-group" data-username="comprog-ux" data-post="1" data-topic="169559">
<div class="title">
<div class="quote-controls"></div>
 comprog-ux:</div>
<blockquote>
<pre><code class="lang-auto">for (i = 1; i &lt; locations; i++) {
var locations = JSON.parse('{{$data}}');
</code></pre>
</blockquote>
</aside>

**Comment by comprog-ux** [comment, marked as answer]
<p>I found out the way to do it.<br>
var locations = [<br>
<span class="mention">@foreach</span> ($data as $i)<br>
[ "{{ $i-&gt;name }}  " + “i " + “{{ $i-&gt;contact_city }}” +  " br&gt;” + “{{ $i-&gt;date }}” + "   Anmälda: " + “{{ $i-&gt;signups_count }}” , {{ $i-&gt;lat }},{{ $i-&gt;lng }},0, “{{ $i-&gt;signups_closing_date }}” , “{{ $i-&gt;closed_at }}”,‘<a href="http://localhost:8080/app/competitions/'+" rel="noopener nofollow ugc">http://localhost:8080/app/competitions/’+</a>{{ $i-&gt;id }}+ ‘/information’],<br>
<span class="mention">@endforeach</span><br>
];<br>
And for the loop where I put out the markers:<br>
var t=new Date();<br>
var visa;<br>
var ldstr=t.toLocaleDateString(); // Today’s date.<br>
// The loop<br>
for (i = 0; i &lt; locations.length; i++) {<br>
date1=new Date(locations[i][4]);<br>
date2=new Date(ldstr);<br>
var newLat = locations[i][1] + (Math.random() -.5) / 20;// * (Math.random() * (max - min) + min);<br>
var newLng = locations[i][2] + (Math.random() -.5) / 20;// * (Math.random() * (max - min) + min);<br>
if (locations[i][5]){<br>
var marker = new L.marker([newLat, newLng],marker3);  // closed competition<br>
marker.url = locations[i][6]; // pick up the date for closed competition<br>
marker.bindPopup(locations[i][0]+’ br&gt; Stängd’);<br>
marker.addTo(map);<br>
marker.on(‘click’, function(){<br>
window.location = (this.url);<br>
});marker.on(‘mouseover’, function (e) {this.openPopup();});<br>
marker.on(‘mouseout’, function (e) {  this.closePopup();});<br>
} else if ( date1 &lt; date2 ) {<br>
var marker = new L.marker([newLat, newLng], marker2);<br>
and so on…<br>
The random lines are to separate competitions for a couple of kilometers on the same city as you can see on the map (cluster).<br>
I have cut the &lt; off the br-s for better reading…<br>
Works perfect. Thanks guys!</p>

**Comment by comprog-ux** [comment]
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/e/e/eecf024116e8e59940882f622ea236f0b4e086ba.png" data-download-href="/uploads/short-url/y4B9rooDnWVuESfqWG1VdcYl3T4.png?dl=1" title="Skärmavbild 2021-04-23 kl. 14.22.53" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/72607670/181064627-4d2c25b9-fb4b-4c00-b534-b811dd889661.png" alt="Skärmavbild 2021-04-23 kl. 14.22.53" data-base62-sha1="y4B9rooDnWVuESfqWG1VdcYl3T4" width="586" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/e/e/eecf024116e8e59940882f622ea236f0b4e086ba_2_586x500.png, https://aws1.discourse-cdn.com/github/optimized/3X/e/e/eecf024116e8e59940882f622ea236f0b4e086ba_2_879x750.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/e/e/eecf024116e8e59940882f622ea236f0b4e086ba.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/e/e/eecf024116e8e59940882f622ea236f0b4e086ba_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Skärmavbild 2021-04-23 kl. 14.22.53</span><span class="informations">1029×877 592 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
